modules/admin-ui/src/main/resources/web/admin/apps/content-manager/js/app/create/NewContentDialog.ts

```diff
@@ -264,64 +266,145 @@ module app.create {
         }
 
         private loadContentTypes() {
+
+            this.showLoadingMasks();
+
+            wemQ.all(this.prepareRequestsToFetchContentData())
+                .spread((contentTypes: ContentTypeSummary[], directChilds: api.content.ContentResponse<api.content.ContentSummary>,
+                         parentSite: Site) => {
+
+                    this.listItems = this.createListOfContentTypeItems(contentTypes, parentSite);
+                    this.mostPopularItems = this.findMostPopularContentTypes(2, contentTypes, directChilds.getContents());
+
+                    this.resetNewContentDialogContent();
+                    this.toggleMostPopularBlockShown();
+                }).catch((reason: any) => {
+
+                    api.DefaultErrorHandler.handle(reason);
+
+                }).finally(() => {
+                    this.filterList();
+                    this.hideLoadingMasks();
+                    this.handleModalDialogAnimation();
+                }).done();
+        }
+
+        private showLoadingMasks() {
             this.contentList.insertChild(this.contentListMask, 0);
             this.recentList.insertChild(this.recentListMask, 0);
             this.contentListMask.show();
             this.recentListMask.show();
+        }
+
+        private hideLoadingMasks() {
+            this.contentListMask.hide();
+            this.recentListMask.hide();
+        }
 
+        private prepareRequestsToFetchContentData(): wemQ.Promise<any>[] {
             var requests: wemQ.Promise<any>[] = [];
             requests.push(new GetAllContentTypesRequest().sendAndParse());
             if (this.parentContent) {
+                requests.push(new ListContentByPathRequest(this.parentContent.getPath()).sendAndParse());
```

Comment:
There are a lot of checks for this.parentContent not a null. Maybe we can set this.parentContent to Root by default, then be able to decrease their number and just switch it 'not-root' check